keps/sig-cloud-provider/20190422-cloud-controller-manager-migration.md

```diff
@@ -152,53 +160,125 @@ will prevent any of the v1.18 CCMs from claiming the lock. When the current hold
 
 ### Implementation Details/Notes/Constraints [optional]
 
-TODO after KEP summary and proposal is approved
+#### Migration Configuration
 
-### Risks and Mitigations
+The migration lock will be configured by defining new API types that will then be passed into the KCM and CCM.
 
-TODO after KEP summary and proposal is approved
+```go
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// LeaderMigrationConfiguration provides versioned configuration for all migrating leader locks.
+type LeaderMigrationConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
 
-## Design Details
+	// LeaderName is the name of the resource under which the controllers should be run.
+	LeaderName string `json:"leaderName"`
 
-### Test Plan
+	// ControllerLeaders contains a list of migrating leader lock configurations
+	ControllerLeaders []ControllerLeaderConfiguration `json:"controllerLeaders"`
+}
 
-TODO after KEP summary and proposal is approved
+// ControllerLeaderConfiguration provides the configuration for a migrating leader lock.
+type ControllerLeaderConfiguration struct {
+	// Name is the name of the controller being migrated
+	// E.g. service-controller, route-controller, cloud-node-controller, etc
+	Name string `json:"name"`
 
-### Graduation Criteria
+	// Component is the name of the component in which the controller should be running.
+	// E.g. kube-controller-manager, cloud-controller-manager, etc
+	Component string `json:"component"`
```

Comment:
If we are calling this leader elsewhere, then we should probably call it leader here.